# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/gamepad/ipc/GamepadEventChannelParent.h
# Commit: 8132dd939023
# Full Hash: 8132dd9390233328c8da7072f36859424a2b079d
# Author: Chris Martin <cmartin@mozilla.com>
# Date: 2020-10-14 21:42:48
# Regressor Bug: 1657404
# File Overlap Count: 2
# Description:
#   Bug 1657404 - Change PGamepadEventChannel to "refcounted protocol" r=handyman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D93119
# ==============================================================================

diff -r 186ca9e93e1d -r 8132dd939023 dom/gamepad/ipc/GamepadEventChannelParent.h
--- a/dom/gamepad/ipc/GamepadEventChannelParent.h	Wed Oct 07 11:34:02 2020 +0000
+++ b/dom/gamepad/ipc/GamepadEventChannelParent.h	Wed Oct 14 15:58:36 2020 +0000
@@ -13,10 +13,9 @@
 
 class GamepadEventChannelParent final : public PGamepadEventChannelParent {
  public:
-  NS_INLINE_DECL_THREADSAFE_REFCOUNTING(GamepadEventChannelParent)
-  GamepadEventChannelParent() = default;
+  NS_INLINE_DECL_THREADSAFE_REFCOUNTING(GamepadEventChannelParent, override)
 
-  bool Init();
+  static already_AddRefed<GamepadEventChannelParent> Create();
   void ActorDestroy(ActorDestroyReason aWhy) override;
 
   mozilla::ipc::IPCResult RecvVibrateHaptic(const uint32_t& aControllerIdx,
@@ -31,8 +30,16 @@
       const uint32_t& aPromiseID);
   void DispatchUpdateEvent(const GamepadChangeEvent& aEvent);
 
+  GamepadEventChannelParent(const GamepadEventChannelParent&) = delete;
+  GamepadEventChannelParent(GamepadEventChannelParent&&) = delete;
+  GamepadEventChannelParent& operator=(const GamepadEventChannelParent&) =
+      delete;
+  GamepadEventChannelParent& operator=(GamepadEventChannelParent&&) = delete;
+
  private:
+  GamepadEventChannelParent();
   ~GamepadEventChannelParent() = default;
+
   nsCOMPtr<nsIEventTarget> mBackgroundEventTarget;
 };
 